apps/api/src/server/repositories/s51-advice-document.repository.js

```diff
@@ -74,3 +74,25 @@ export const getPublishedDocumentsByAdviceIds = (s51AdviceIds) => {
 		}
 	});
 };
+
+/**
+ *
+ * @param {number} s51AdviceId
+ * @param {string} documentId
+ * @returns {Promise<*>}
+ */
+export const upsertS51AdviceDocument = async (s51AdviceId, documentId) => {
+	const transactionItems = [
+		databaseConnector.s51AdviceDocument.upsert({
+			where: { documentGuid: documentId },
+			update: { adviceId: s51AdviceId },
+			create: {
+				documentGuid: documentId,
+				adviceId: s51AdviceId
+			}
+		})
+	];
+
+	const [upsertResult] = await databaseConnector.$transaction(transactionItems);
```

Comment:
I don't think we need transaction as its just one prisma query

Reply:
Fixed!